src/addons/addonmessage.cpp

```diff
@@ -101,11 +101,12 @@ void AddonMessage::updateMessageState(State newState) {
 
   QMetaEnum stateMetaEnum = QMetaEnum::fromType<State>();
   QString newStateSetting = stateMetaEnum.valueToKey(newState);
-  SettingsHolder* settingsHolder = SettingsHolder::instance();
-  settingsHolder->setAddonSetting(StateQuery(id()), newStateSetting);
 
   m_state = newState;
   emit stateChanged(m_state);
+
```

Comment:
This is needed to receive the `addonSettingsChanged` after `stateChanged`.

Reply:
I think one of the reasons this was broken by my PR is that there no tests to guarantee the behaviour. Is it possible to write tests for it? If so, would you prefer to do that on this PR or file a follow up?